### ISSUE TITLE
You cannot combine: --no-cache --fail-on-change

### ISSUE THREAD

**Issue by terlar** [numtide/treefmt#343]
**Describe the bug**

If you use `--no-cache`, the option `--fail-on-change` won't have any effect.

**To Reproduce**

1. `git init`
2. Create a file `no-cache-fail-on-change-bug.yaml`:
```yaml
key1: value1

key2: value2


key3: value3
```
3. `git add no-cache-fail-on-change-bug.yaml`
4. Configure treefmt with `yamlfmt`
5. Run `treefmt --no-cache --fail-on-change`
6. Observe no errors, but the file has changed
7. `git checkout no-cache-fail-on-change-bug.yaml`
8. Repeat steps with `treefmt --fail-on-change`
9. Observe `treefmt: error: unexpected changes detected, --fail-on-change is enabled`

**Expected behavior**
When `--fail-on-change` is used together with `--no-cache` it should work.

**System information**

Linux

```
$ treefmt --version
treefmt v2.0.2
```

**Comment by brianmcgee** [comment]
@terlar yeah looks like https://github.com/numtide/treefmt/pull/330 broke this, I'm having a look

**Comment by brianmcgee** [comment]
@terlar I have a fix inbound #344 

I'll probably cut a new release over the weekend, as there will be a few fixes and improvements. 

In the meantime, you can pull in the branch as a flake input and override the package being used.

**Comment by terlar** [comment]
I can confirm that this works now
